src/auth/Auth.ts

```diff
@@ -0,0 +1,100 @@
+/* eslint-disable @typescript-eslint/no-unused-vars -- This file contains a base class with a lot
+   of unused paramters. */
+
+import type { ConnectOptions, Credentials } from "@canonical/jujulib";
+
+import type { ConnectionWithFacades } from "juju/types";
+import type { AuthMethod, Credential } from "store/general/types";
+import type { AppDispatch } from "store/store";
+import { logger } from "utils/logger";
+
+/**
+ * Base class for authentication providers. The stubbed lifecycle methods are called throughout the
+ * dashboard. Authentication implementations should extend this class and override the relevant
+ * methods.
+ *
+ * This class should be consumed as a singleton via the `instance` property (eg.
+ * `Auth.instance.logout()`). The singleton is set in the base class's constructor.
+ */
+export class Auth {
+  /**
+   * Singleton instance of the auth provider.
+   */
+  static instance: Auth;
+
+  /**
+   * Redux dispatch method for use within auth implementations.
+   */
+  protected dispatch: AppDispatch;
+
+  /**
+   * Identifier of the auth method.
+   */
+  public name: AuthMethod;
+
+  /**
+   * Create a new instance of the singleton, using the
+   */
+  constructor(dispatch: AppDispatch, name: AuthMethod) {
+    if (Auth.instance) {
+      logger.warn(
+        "Singleton instance already exists, and is being re-constructed. Overriding previous instance with incoming.",
+      );
+    }
+
+    Auth.instance = this;
+    this.dispatch = dispatch;
+    this.name = name;
+  }
```

Comment:
It would be good to add tests for the initialised class and that the instance is created (don't need a test at the logger is called though I don't think).